rfcs/0058-secrets-for-services.md

```diff
@@ -0,0 +1,273 @@
+---
+feature: secrets_for_services
+start-date: 2019-10-29
+author: @d-goldin
+co-authors: (find a buddy later to help our with the RFC)
+
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+This RFC introduces some interfaces, terminology and library functions to help managing
+secrets for NixOS systemd services modules.
+
+The general idea is to provide some basic infrastructure within nixos modules to
+handle secrets more consistently while being able to integrate pre-existing solutions
+like NixOps, or a simple secrets folder.
+
+# Motivation
+[motivation]: #motivation
+
+There is currently a lack of consistent and safe mechanisms to make secrets
+available to systemd services in NixOS. Various modules implement it in various
+ways across the ecosystem. There have also been ideas like adjustments to the
+Nix Store (like issue https://github.com/NixOS/nixpkgs/issues/8), which
```

Comment:
This should be https://github.com/NixOS/nix/issues/8.

Reply:
Whoops. Thanks!